### ISSUE TITLE
Upgrade all libraries to latest versions

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#1247]
We should try to upgrade all the libraries to latest versions.

I did a bit of research on some of them already:
 - [x] axios -> we should upgrade it to 1.4.0. Migration should be none or very little. Check here: https://github.com/bmuenzenmeyer/axios-1.0.0-migration-guide/ .
 - [ ] @tanstack/react-query: [#2244](https://github.com/wasp-lang/wasp/issues/2244)
 - [ ] react-router-dom -> #2245 
   Also upgrade @types/react-router-dom together with it.
 - [x] typescript -> upgrade to 5.0.2 .

Make sure to check others also!

**Comment by sodic** [comment]
React router causing problems for users: https://discord.com/channels/686873244791210014/1263461365872595054/1263480810342711337

> I don't know why, but it seems the React Router version is v5.x.x, so it doesn't have useNavigate. I can only use useHistory

**Comment by cprecioso** [comment]
@Martinsos should we close in favor of https://github.com/wasp-lang/wasp/issues/2886 ?

**Comment by Martinsos** [comment]
@cprecioso yes I thikn so! Move any info that might be beneficial, and then feel free to close.
Maybe it would be good if I just turned this issue into epic instead of creating new one, but ok no big deal.
